src/components/PublicCommentCard/PublicCommentCard.tsx

```diff
@@ -0,0 +1,146 @@
+export const PublicCommentCard = ({
+  comment,
+  commentNumber = 0,
+}: PublicCommentCardProps) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  if (!comment) {
+    return (
+      <div className="dpr-public-comment-card">
+        <div className="dpr-public-comment-card__header">
+          <div className="dpr-public-comment-card__skeleton--item" />
+          <div className="dpr-public-comment-card__skeleton--title" />
+          <div className="dpr-public-comment-card__skeleton--body" />
+        </div>
+      </div>
+    );
+  }
+
+  const topicsAndComments: TopicAndComments[] = Array.isArray(comment.comment)
+    ? comment.comment
+    : [
+        {
+          topic: "other",
+          question: "",
+          comment: comment.comment as string,
+        },
+      ];
+
+  const collapsedTopicsAndComments =
+    collapseTopicsByCharLimit(topicsAndComments);
+
+  const displayedTopicsAndComments = isExpanded
+    ? topicsAndComments.map((t, i) => ({
+        originalIndex: i,
+        topic: t.topic,
+        question: t.question,
+        comment: t.comment,
+        truncated: false,
+      }))
+    : collapsedTopicsAndComments;
+
+  const hasOverflow =
+    collapsedTopicsAndComments.length < topicsAndComments.length ||
+    collapsedTopicsAndComments.some((topicObj) => topicObj.truncated);
+
+  const commentId = comment.id ?? commentNumber;
+
+  return (
+    <div className="dpr-public-comment-card">
+      <div
+        className="dpr-public-comment-card__header"
+        id={`public-comment-${commentId}`}
+      >
+        <h3 className="govuk-heading-m">{`Comment #${commentId}`}</h3>
+        {comment.metadata?.publishedAt && (
+          <p className="govuk-body">
+            <em>
+              Published{" "}
+              <time dateTime={comment.metadata.publishedAt}>
+                {formatDateTimeToDprDate(comment.metadata.publishedAt)}
+              </time>
+            </em>
+          </p>
+        )}
+
+        {comment.sentiment && (
+          <div>
+            <div className="govuk-heading-s">
+              Sentiment towards this application
+            </div>
+            <p className="govuk-body">{capitaliseWord(comment.sentiment)}</p>
+          </div>
+        )}
+
+        {displayedTopicsAndComments.map((topicObj) => {
+          const option = COMMENT_PUBLIC_TOPIC_OPTIONS.find(
+            (o) => o.value === topicObj.topic,
+          );
+          const title = option?.label ?? capitaliseWord(topicObj.topic);
+
+          return (
+            <div
+              key={topicObj.originalIndex}
+              className="dpr-public-comment-card__topic-section"
+            >
+              <div className="govuk-heading-s">{title}</div>
+              <div className="govuk-body">
+                {topicObj.comment}
+                {!isExpanded && topicObj.truncated && "…"}
+              </div>
+            </div>
+          );
+        })}
+
+        {hasOverflow && (
+          <button
```

Comment:
Should this be `TextButton`? rn its 
<img width="294" alt="Screenshot 2025-05-28 at 10 26 23 pm" src="https://github.com/user-attachments/assets/0ccad27d-b359-4fb7-8116-1d20e06ca3dd" />